.github/workflows/docker-images.yml

```diff
@@ -0,0 +1,65 @@
+name: Generate Docker Images
+# TODO: it's triggered on PRs for QA, but it'll be on pushes to master when it's
+# done.
```

Comment:
Quick question - is this for deployment or for development? Where do the images go?

Reply:
It's for deployment.  The artifact should be available through [GitHub itself](https://github.com/actions/upload-artifact#where-does-the-upload-go), but the intention is to link it to whatever we use for deployment (TBD).

Reply:
[Tada!](https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/767447858) (at the bottom)

Reply:
Nice!!!

So my next question is - do we want this to run on merge to `main` or on merge to our production branches?

Reply:
I'm not 100% sure, but probably on merges to `main`.  As I see it the deployment sequence will end up being something like this:

1. build images, triggered by merge to `main`
2. run CI tests using those images
3. if they pass, push those images to staging VMs
4. if those are okay, repeat for production VMs

perhaps we'll still want to use production branches with this, but it's a different flow, so maybe not.